### PR TITLE
Dit 79 - View BTI applications and rulings on Trader UI

### PR DESCRIPTION
**Pull request by pllaurado**
Trader can view a list of all their BTI applications and rulings

### CODE REVIEW

File: app/connectors/BindingTariffClassificationConnector.scala [hunk 1]
```diff
@@ -39,4 +40,12 @@ class BindingTariffClassificationConnector @Inject()(configuration: FrontendAppC
     val url = s"${configuration.bindingTariffClassificationUrl}/cases/$reference"
     client.GET[Option[Case]](url)
   }
+
+  def findCasesBy(eori: String, status: Seq[CaseStatus], pagination: Pagination)(implicit hc: HeaderCarrier): Future[Paged[Case]] = {
+    val url = s"${configuration.bindingTariffClassificationUrl}/cases?eori=$eori&status=${status.mkString(",")}" +
+      s"&sort_by=decision-end-date&sort_direction=desc" +
```
Comment (ed0906) [inline app/connectors/BindingTariffClassificationConnector.scala]: Do we have this sort by field?

File: app/controllers/IndexController.scala [hunk 2]
```diff
@@ -16,19 +16,40 @@
...
+import scala.concurrent.Future.successful
+
 class IndexController @Inject()(val appConfig: FrontendAppConfig,
+                                identify: IdentifierAction,
+                                service: CasesService,
                                 val messagesApi: MessagesApi) extends FrontendController with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = Action { implicit request =>
-    Ok(index(appConfig))
+  def loadApplications(page : Int): Action[AnyContent] = identify.async { implicit request =>
```
Comment (ed0906) [inline app/controllers/IndexController.scala]: `load` seems a little strange, i'd call it `getRulings` and `getApplications`

File: app/models/Cancellation.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.CancelReason.CancelReason
+
+case class Cancellation
```
Comment (ed0906) [inline app/models/Cancellation.scala]: Does the Trader UI care about cancellations?

File: app/utils/Dates.scala [hunk 4]
```diff
@@ -27,4 +27,8 @@ object Dates {
     formatter.format(LocalDateTime.ofInstant(instant, ZoneOffset.UTC))
   }
 
+  def format(instant: Option[Instant]) : String = {
+    instant.map( in => formatter.format(LocalDateTime.ofInstant(in , ZoneOffset.UTC))).getOrElse("None")
+  }
```
Comment (SteveSugden) [inline app/utils/Dates.scala]: why does this format method not call the other one? DRY?

File: app/models/Appeal.scala [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.AppealStatus.AppealStatus
+
+case class Appeal
```
Comment (ed0906) [inline app/models/Appeal.scala]: Does the trader UI care about appeals?

File: app/models/CancelReason.scala [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+object CancelReason extends Enumeration {
```
Comment (ed0906) [inline app/models/CancelReason.scala]: Does the trader ui care about cancelation reasons?

File: conf/messages.en [hunk 7]
```diff
@@ -336,6 +336,8 @@ confirmation.paragraph.whatNext0.sample = We will give you your ruling within 30
 confirmation.paragraph.whatNext0.nosample = We will give you your ruling within 30 to 60 days.
 confirmation.paragraph.whatNext1 = You can sign in to check the status of your application.
 
+confirmation.refValue = GB{0}
```
Comment (SteveSugden) [inline conf/messages.en]: Why is this back?  GB prefix is no more and we are now using a common empty format for confirmation page and application pdf

File: app/models/Decision.scala [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import java.time.Instant
+
+case class Decision
```
Comment (ed0906) [inline app/models/Decision.scala]: We should only include the fields we actually need

File: app/models/Pagination.scala [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+object Pagination {
+  val unlimited: Int = Integer.MAX_VALUE
+}
+
+case class SearchPagination
+(
+  override val page: Int = 1,
+  override val pageSize: Int = 10
```
Comment (ed0906) [inline app/models/Pagination.scala]: Only 10?

File: app/models/Review.scala [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.ReviewStatus.ReviewStatus
+
+
+case class Review
```
Comment (ed0906) [inline app/models/Review.scala]: Does the trader ui care about reviews?

File: app/models/ReviewStatus.scala [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+object ReviewStatus extends Enumeration {
```
Comment (ed0906) [inline app/models/ReviewStatus.scala]: Does the trader ui care about review status?

File: app/utils/Status.scala [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+  def applicationFormat: CaseStatus => String = {
+    case CaseStatus.NEW | CaseStatus.OPEN | CaseStatus.SUSPENDED => "In progress"
+    case CaseStatus.SUPPRESSED | CaseStatus.REJECTED => "Rejected"
+    case CaseStatus.REFERRED => "Info requested"
+    case CaseStatus.COMPLETED => "Completed"
+    case _ => "Unknown"
+  }
+
+  def hasActiveDecision = { c: Case =>
```
Comment (ed0906) [inline app/utils/Status.scala]: Isnt this on the case now?

File: app/utils/Status.scala [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    c.decision.flatMap(_.effectiveEndDate).exists(_.compareTo(Instant.now) >= 0)
+  }
+
+  def hasExpiredDecision = { c: Case =>
```
Comment (ed0906) [inline app/utils/Status.scala]: Isnt this on the case now?

File: app/views/PaginationUtil.scala [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    var indexes: Seq[Int] = Seq(pageIndex)
+    var surrounded = surround(indexes, pageCount)
+    while(surrounded.size <= available && indexes != surrounded) {
+      indexes = surrounded
+      surrounded = surround(indexes, pageCount)
+    }
+    indexes
+  }
+
+  private def surround(list: Seq[Int], pageCount: Int): Seq[Int] = {
```
Comment (ed0906) [inline app/views/PaginationUtil.scala]: Look at this beautiful code

File: app/connectors/BindingTariffClassificationConnector.scala (hunk 1 above)
Comment (pllaurado) [reply app/connectors/BindingTariffClassificationConnector.scala]: yes

File: conf/messages.en (hunk 7 above)
Comment (pllaurado) [reply conf/messages.en]: I have no idea what is this value.... Maybe the merge from master

File: app/models/Appeal.scala (hunk 5 above)
Comment (ed0906) [reply app/models/Appeal.scala]: I'll take that as a no ;)

File: conf/messages.en (hunk 7 above)
Comment (ed0906) [reply conf/messages.en]: reference no tiene `GB`

File: app/models/Pagination.scala (hunk 9 above)
Comment (pllaurado) [reply app/models/Pagination.scala]: 50

File: app/models/Review.scala (hunk 10 above)
Comment (pllaurado) [reply app/models/Review.scala]: No

File: app/utils/Status.scala (hunk 12 above)
Comment (pllaurado) [reply app/utils/Status.scala]: this case should me removed

File: app/views/PaginationUtil.scala (hunk 14 above)
Comment (pllaurado) [reply app/views/PaginationUtil.scala]: I was want to mention that I was a bit weak when I wrote this

File: app/models/Appeal.scala (hunk 5 above)
Comment (pllaurado) [reply app/models/Appeal.scala]: exactly ;)

File: app/views/PaginationUtil.scala [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+package views
+
+import scala.collection.immutable.Seq
+
+object PaginationUtil {
+
+  def surroundingPages(pageIndex: Int, pageCount: Int, available: Int): Seq[Int] = {
+    var indexes: Seq[Int] = Seq(pageIndex)
```
Comment (gokyo) [inline app/views/PaginationUtil.scala]: `var` ?

**Comment by ed0906** [comment]
Looks good, feel free to merge

File: app/views/PaginationUtil.scala (hunk 15 above)
Comment (ed0906) [reply app/views/PaginationUtil.scala]: It’s required :)

**Pull request merged**